### PR TITLE
extract case update code to separate class

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph 

Just moving code around. Extracting the code that updates a case into its own class.

This is a PR into #8644 

cc @millerdev

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/update_strategy.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+    @classmethod
+    def case_from_case_update(cls, case_update, xformdoc):
+        """
+        Create a case object from a case update object.
+        """
+        assert not is_deprecation(xformdoc)  # you should never be able to create a case from a deleted update
+        case = CommCareCase()
+        case._id = case_update.id
+        case.modified_on = parsing.string_to_utc_datetime(case_update.modified_on_str) \
+                            if case_update.modified_on_str else datetime.utcnow()
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/update_strategy.py]: E127 continuation line over-indented for visual indent

File: corehq/ex-submodules/casexml/apps/case/update_strategy.py [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+        self.soft_rebuild_case(strict=False, xforms=local_forms)
+
+        if case_update.version:
+            self.case.version = case_update.version
+
+    def reset_case_state(self):
+        """
+        Clear known case properties, and all dynamic properties
+        """
+        dynamic_properties = set([k for action in self.case.actions for k in action.updated_unknown_properties.keys()])
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/update_strategy.py]: E501 line too long (119 > 115 characters)

**Comment by millerdev** [comment]
:+1: nice cleanup

**Comment by benrudolph** [comment]
makes sense nice change

**Comment by benrudolph** [comment]
(unmergable)

**Comment by czue** [comment]
:walking: :walking:

**Comment by czue** [comment]
(this is great)

**Comment by snopoke** [comment]
closed in favour of https://github.com/dimagi/commcare-hq/pull/8667

**Pull request closed without merging**